src/components/cc-addon-linked-apps/cc-addon-linked-apps.smart.js

```diff
@@ -79,11 +79,12 @@ function fetchApplications({ apiConfig, signal, ownerId, addonId }) {
 /**
  * @param {Object} parameters
  * @param {ApiConfig} parameters.apiConfig
+ * @param {string} parameters.ownerId
  * @param {AbortSignal} parameters.signal
  * @returns {Promise<Zone[]>}
  */
-function fetchZones({ apiConfig, signal }) {
-  return getAllZones().then(sendToApi({ apiConfig, signal, cacheDelay: ONE_DAY }));
+function fetchZones({ apiConfig, signal, ownerId }) {
+  return listZones({ ownerId }).then(sendToApi({ apiConfig, signal, cacheDelay: ONE_DAY }));
```

Comment:
issue: we can't use cacheDelay anymore as we might not retrieve the zones for other orgas if you switched in the console.

Reply:
Wow, thank you, I'll fix this. :brain:

Reply:
@Galimede doesn't the cache system provided by `sendToApi` take into account the `queryParams`? I think they are part of the cache key and if they are.

If `queryParams` are part of the cache key, I think it's fine and we'll only rely on the client cache if we've made a call with this specific `ownerId` before :thinking: (not sure, that's just how I understand the `sendToApi` code)

Reply:
Yeah you're right, that's my bad, nice catch! :hugs:

Reply:
TIL. Thanks to both of you, I'll fix this!